dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationResultImpl.java

```diff
@@ -7,37 +7,47 @@
 package com.powsybl.dynamicsimulation;
 
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
-import com.powsybl.timeseries.StringTimeSeries;
 import com.powsybl.timeseries.TimeSeries;
 
 /**
  * @author Marcos de Miguel {@literal <demiguelm at aia.es>}
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
  */
 public class DynamicSimulationResultImpl implements DynamicSimulationResult {
 
-    private final boolean status;
-    private final String logs;
+    private final Status status;
+    private final String error;
     private final Map<String, TimeSeries> curves;
-    private final StringTimeSeries timeLine;
+    private final List<TimelineEvent> timeLine;
 
-    public DynamicSimulationResultImpl(boolean status, String logs, Map<String, TimeSeries> curves, StringTimeSeries timeLine) {
-        this.status = status;
-        this.logs = logs;
+    public DynamicSimulationResultImpl(Status status, String error, Map<String, TimeSeries> curves, List<TimelineEvent> timeLine) {
+        this.status = Objects.requireNonNull(status);
+        this.error = error;
```

Comment:
I think we could also require that input to be non-null.

We could also replace the null input with the name of the input, as it's done on `LoadFlowResultImpl`: `Objects.requireNonNullElse(statusText, status.name());`, but I'm not so much convinced by adding this useless info, are you?